[Backend] Implemented input validation for post requests

amlannandy opened the pull request:
**Description**

Implemented input validation in post routes by adding the following measures -
1. If not post data is sent at all
2. If some data is sent but not the fields that are required

**Demo**

![screen-capture (1)](https://user-images.githubusercontent.com/45410599/108306543-623a7080-71d2-11eb-9a42-dc1d005677d5.gif)

fixes #71

amlannandy committed:
Add input validation for post requests

shivanshuraj1333 commented:
@amlannandy please resolve conflicts

amlannandy committed:
Merge branch 'master' into validate-post-req

amlannandy commented:
@shivanshu1333 Done. As we're now only deleting the current user, email is not necessary. So I removed the input validation.

shivanshuraj1333 commented on flask-backend/api/routes/user.py:
```diff
@@ -105,15 +105,6 @@ def roleupdate():
-
-    user = User.query.filter_by(email=email).first()
-    db.session.delete(user)
+    db.session.delete(current_user)
```
this API will be used by the admin to delete any particular user, we're not supposed to delete the current user here.

amlannandy replied:
@shivanshu1333 Yeah that's what I was also thinking. You approved #68  so I thought we're changing the functionality. I'll change it back.

sachinsom93 replied:
Hi @shivanshu1333, @amlannandy, 
Previously I also thought that only admin can delete any of his user, but at that time some funtionality related to admin was not created like create user by admin, route to show all users of admin, so thought to change delete function in such a way that only a user can delete only his account as previously any user can delete anybody's account.
I have also mentioned a issue related to this, I'll soon raise a PR for that, it would contain route for only admin can create users and delete his users.

amlannandy replied:
@sachinsom93 I see. Sounds good. 👍🏻

amlannandy committed:
fixed delete user route

amlannandy commented:
@shivanshu1333 I have changed the delete route. Do let me know if it works or if any other change is required.

sachinsom93 commented:
> @shivanshu1333 I have changed the delete route. Do let me know if it works or if any other change is required.

Hi @amlannandy, I have raised PR #76 for admin related funtionality where I've created a "remove-user" route for  deleting a user by his admin, so I think we don't need this delete route. We have to wait for @shivanshu1333 's approval for PR #76.

amlannandy commented:
> > @shivanshu1333 I have changed the delete route. Do let me know if it works or if any other change is required.
> 
> Hi @amlannandy, I have raised PR #76 for admin related funtionality where I've created a "remove-user" route for deleting a user by his admin, so I think we don't need this delete route. We have to wait for @shivanshu1333 's approval for PR #76.

Gotcha. I have just reverted the change as I was asked. Will undo the last commit if needed. :)

shivanshuraj1333 commented:
Please resolve conflicts

amlannandy committed:
Merge branch 'master' into validate-post-req

amlannandy commented:
> Please resolve conflicts

Done 👍🏻
